addons/mail/tools/discuss.py

```diff
@@ -200,7 +202,12 @@ def many_ids(records, mode="REPLACE", /, *, as_thread=False):
         should be avoided. It is kept as a public method until all remaining
         occurences can be removed.
         Using the method ``many(..., only_id=True)`` is preferable."""
-        res = [Store.one_id(record, as_thread=as_thread) for record in records]
+        if records._name == "mail.message.reaction":
+            res = []
+            for (message_id, content), _ in groupby(records, lambda r: (r.message_id, r.content)):
+                res.append({"message": {"id": message_id.id}, "content": content})
```

Comment:
more python/efficient syntax (also rename `message_id` as it's actually a recordset there)
```suggestion
            res = [
                {"message": {"id": message.id}, "content": content}
                for (message, content), _ in groupby(records, lambda r: (r.message_id, r.content))
            ]
```